From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nassim Jahnke <nassim@njahnke.dev>
Date: Wed, 6 Dec 2023 21:24:54 +0100
Subject: [PATCH] HACK id string codec in hover event

Temporary hack to remove serialization to the int array, since adventure has not been updated to deserialize this format


diff --git a/src/main/java/net/minecraft/network/chat/HoverEvent.java b/src/main/java/net/minecraft/network/chat/HoverEvent.java
index 7fd85ae2ebd7225f06d874aa7e37fbdb89e3ea92..7ff694a3dd492224dedfa00b4ff0ff3015b50d1c 100644
--- a/src/main/java/net/minecraft/network/chat/HoverEvent.java
+++ b/src/main/java/net/minecraft/network/chat/HoverEvent.java
@@ -134,7 +134,7 @@ public class HoverEvent {
         public static final Codec<HoverEvent.EntityTooltipInfo> CODEC = RecordCodecBuilder.create((instance) -> {
             return instance.group(BuiltInRegistries.ENTITY_TYPE.byNameCodec().fieldOf("type").forGetter((content) -> {
                 return content.type;
-            }), UUIDUtil.LENIENT_CODEC.fieldOf("id").forGetter((content) -> {
+            }), UUIDUtil.STRING_CODEC.fieldOf("id").forGetter((content) -> {
                 return content.id;
             }), ExtraCodecs.strictOptionalField(ComponentSerialization.CODEC, "name").forGetter((content) -> {
                 return content.name;
